### PR TITLE
fix(sdcm/nemesis.py): ToggleTableICS: use command of alter materialized view for MV table

### LINKED ISSUE #1730: ToggleTableICS nemesis not using 'ALTER MATERIALIZED VIEW', failing with:  "Cannot use ALTER TABLE on Materialized View
the nemesis should use 'alter table' command for standard tables and 'alter materialized view' for MV tables.
failure:
```
16:29:10  < t:2020-01-23 14:29:09,756 f:nemesis.py      l:636  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.GeminiNonDisruptiveChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method toggle_table_ics
16:29:10  < t:2020-01-23 14:29:09,756 f:nemesis.py      l:385  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.GeminiNonDisruptiveChaosMonkey: ToggleTableICS
16:29:10  < t:2020-01-23 14:29:09,767 f:sct_events.py   l:639  c:sdcm.sct_events      p:INFO  > 2020-01-23 14:29:09.756: (DisruptionEvent Severity.NORMAL): type=start name=ToggleTableICS node=Node ics-gemini-nemesis-3h-gemini-3-db-node-528255c4-2 [63.32.91.15 | 10.0.89.97] (seed: False) duration=None
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:642  c:sdcm.nemesis         p:ERROR > sdcm.nemesis.GeminiNonDisruptiveChaosMonkey: ('Exception in random_disrupt_method %s: %s', 'toggle_table_ics', <UnexpectedExit: cmd='cqlsh --no-color   --request-timeout=120 --connect-timeout=5  -e "ALTER TABLE ks1.table1_mv_0 WITH compaction = {\'class\': \'IncrementalCompactionStrategy\', \'bucket_high\': 1.75, \'bucket_low\': 0.78, \'min_sstable_size\': 220, \'min_threshold\': 10, \'max_threshold\': 58, \'sstable_size_in_mb\': 1573};" 10.0.89.97 9042' exited=2>)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > sdcm.nemesis.GeminiNonDisruptiveChaosMonkey: Unhandled exception in method <function GeminiNonDisruptiveChaosMonkey.disrupt at 0x7fbd770ecea0> < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > sdcm.nemesis.GeminiNonDisruptiveChaosMonkey: Unhandled exception in method <function GeminiNonDisruptiveChaosMonkey.disrupt at 0x7fbd770ecea0>
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > Traceback (most recent call last):
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 1564, in wrapper
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     result = method(*args, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 2165, in disrupt
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     self.call_random_disrupt_method(disrupt_methods=self.disrupt_methods_list)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 639, in call_random_disrupt_method
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     disrupt_method()
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 1051, in disrupt_toggle_table_ics
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     self.toggle_table_ics()
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 933, in toggle_table_ics
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     raise unexpected_exit
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/nemesis.py", line 929, in toggle_table_ics
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     self.target_node.run_cqlsh(cmd)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/cluster.py", line 2519, in run_cqlsh
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     verbose=verbose)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/remote.py", line 261, in run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     result = _run()
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/utils/common.py", line 92, in inner
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     return func(*args, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/sct/sdcm/remote.py", line 248, in _run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     watchers=watchers, timeout=timeout)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "<decorator-gen-3>", line 2, in run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 30, in opens
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     return method(self, *args, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/fabric/connection.py", line 721, in run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     return self._run(self._remote_runner(), command, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/context.py", line 101, in _run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     return runner.run(command, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 291, in run
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     return self._run_body(command, **kwargs)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >   File "/usr/local/lib/python3.6/site-packages/invoke/runners.py", line 442, in _run_body
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR >     raise UnexpectedExit(result)
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > invoke.exceptions.UnexpectedExit: Encountered a bad command exit code!
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > Command: 'cqlsh --no-color   --request-timeout=120 --connect-timeout=5  -e "ALTER TABLE ks1.table1_mv_0 WITH compaction = {\'class\': \'IncrementalCompactionStrategy\', \'bucket_high\': 1.75, \'bucket_low\': 0.78, \'min_sstable_size\': 220, \'min_threshold\': 10, \'max_threshold\': 58, \'sstable_size_in_mb\': 1573};" 10.0.89.97 9042'
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > Exit code: 2
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > Stdout:
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > Stderr:
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > 
16:29:13  < t:2020-01-23 14:29:12,801 f:nemesis.py      l:1574 c:sdcm.nemesis         p:ERROR > <stdin>:1:InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot use ALTER TABLE on Materialized View"
```

### PR DESCRIPTION
**Pull request by yarongilor**
fixes #1730
fixes https://github.com/scylladb/scylla-cluster-tests/issues/1730

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by bentsi** [comment]
@yarongilor please fix pre-commit issues

**Comment by yarongilor** [comment]
> @yarongilor please fix pre-commit issues

@bentsi , fixed and started a test job in: https://jenkins.scylladb.com/job/scylla-staging/job/longevity-tests-ICS/job/longevity-ics-toggle-strategy-large-partitions-24h_staging/

**Comment by fgelcer** [comment]
need to backport to 4.0, as last lwt short longevity run failed with the issue this PR is fixing.

**Comment by yarongilor** [comment]
@bentsi , is there reason this PR is not merged/reviewed?

**Comment by roydahan** [comment]
@bentsi ?

**Comment by bentsi** [comment]
@yarongilor please resolve conflicts

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -1094,7 +1098,9 @@ def toggle_table_ics(self):
         if new_compaction_strategy in [CompactionStrategy.INCREMENTAL, CompactionStrategy.SIZE_TIERED]:
             for param in list_additional_params:
                 new_compaction_strategy_as_dict.update(param)
-        cmd = "ALTER TABLE {keyspace_table} WITH compaction = {new_compaction_strategy_as_dict};".format(**locals())
+        alter_command_prefix = 'ALTER TABLE ' if keyspace_table not in mview_ks_cfs else 'ALTER MATERIALIZED VIEW '
+        cmd = alter_command_prefix + \
+            " {keyspace_table} WITH compaction = {new_compaction_strategy_as_dict};".format(**locals())
```
Comment (bentsi) [inline sdcm/nemesis.py:1103]: we are in Python 3, use fstrings

Comment (yarongilor) [reply sdcm/nemesis.py:1103]: left the prior mode in order to be compatible with 2019.1 enterprise branch of python-2

**Pull request merged**